app/models/grda_warehouse/source_client_view_accessor.rb

```diff
@@ -15,6 +15,38 @@
 #   end
 #
 class GrdaWarehouse::SourceClientViewAccessor
+  class Cache
```

Comment:
@ttoomey can you add some code comments here to explain what this is doing?  
This feels like a fair amount of abstract logic to shove into a class inside of a class.  Does it make sense to pull this out to its own file?  (Or can we declare it private in some way?)

Reply:
good suggestion, added docs. I guess I would leave that as an interior class if that's not too ugly.

I also replaced the method with a block pattern to make it a little cleaner and hopefully easier to understand

Reply:
Thanks!  I think that change makes this much less daunting.